### PR TITLE
Support freebsd

### LINKED ISSUE #167: Ameba does not install on FreeBSD
Ameba's postinstall script seems to not work on FreeBSD. I initialized an empty project with `crystal init app .`, added the following to `shard.yml`:
```yaml
dependencies:
  ameba:
    github: crystal-ameba/ameba
```
And ran `shards`. Error message:
```
Resolving dependencies
Fetching https://github.com/crystal-ameba/ameba.git
Installing ameba (0.13.2)
Postinstall of ameba: make bin && make run_file
Failed postinstall of ameba on make bin && make run_file:
build
make: exec(build) failed (No such file or directory)
*** Error code 1

Stop.
make: stopped in <project dir>/lib/ameba
```
This is a FreeBSD machine that has GNU make installed from packages (though I suspect the problem is that the FreeBSD package installs GNU make as `gmake` since `make` is BSD make).

### PR DESCRIPTION
**Pull request by veelenga**
- [x] remove incompatible makefile instructions
- [x] added CI step (refs https://github.com/marketplace/actions/freebsd-vm)

duplicates #221
closes #167

image

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -34,3 +34,18 @@ jobs:
...
+    - uses: actions/checkout@v2
+    - name: Install Crystal and test
+      id: test
+      uses: vmactions/freebsd-vm@v0.1.3
+      with:
+        usesh: true
+        prepare: pkg install -y libyaml libevent crystal shards
+        run: |
+          shards install
+          make test
```
Comment (straight-shoota) [inline .github/workflows/ci.yml:51]:
```suggestion
        prepare: pkg install -y libyaml libevent crystal shards gmake
        run: |
          shards install
          gmake test
```

Even if the current makefile works for both, GNU and BSD dialects, it's better to just use GNU make everywhere.
See https://github.com/crystal-ameba/ameba/pull/221#issuecomment-812908622

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -34,3 +34,18 @@ jobs:
...
+    - uses: actions/checkout@v2
+    - name: Install Crystal and test
+      id: test
+      uses: vmactions/freebsd-vm@v0.1.3
+      with:
+        usesh: true
+        prepare: pkg install -y libyaml libevent crystal shards gmake
+        run: |
+          shards install
+          gmake test
```
Comment (Sija) [inline .github/workflows/ci.yml:51]: That would be better fitting under the `matrix.include` key.

Comment (veelenga) [reply .github/workflows/ci.yml:51]: Not sure why that is needed, since the matrix allows to spawn several jobs which is not planned here.

**Comment by veelenga** [comment]
Can't make the freebsd job build stable. Might be some issue with [freebsd-vm](https://github.com/marketplace/actions/freebsd-vm) since it just fails randomly. Closing this in favor of https://github.com/crystal-ameba/ameba/commit/ecafa271ee616b641bc9956b4178ade1b145c573

**Pull request closed without merging**